test/e2e/quick_start_test.go

```diff
@@ -134,17 +132,20 @@ func TestQuickStart(t *testing.T) {
 
 	// Clean up the namespace after the test
 	defer func() {
-		err := input.Client.Delete(ctx, namespace, &crclient.DeleteOptions{})
+		err := input.Client.Delete(TestContext, namespace, &crclient.DeleteOptions{})
 		if err != nil {
 			t.Fatalf("failed to delete namespace %q: %s", namespace.Name, err)
 		}
+		// TODO: given we may have arrived here because TestContext has been closed,
+		// what's a good way to use TestContext if it's still open or a hard-coded
+		// timeout if it's closed?
 		t.Logf("Waiting for the test namespace %q to be deleted", namespace.Name)
 		err = wait.Poll(1*time.Second, 10*time.Minute, func() (done bool, err error) {
 			latestNamespace := &corev1.Namespace{}
 			key := crclient.ObjectKey{
 				Name: namespace.Name,
 			}
-			if err := input.Client.Get(ctx, key, latestNamespace); err != nil {
+			if err := input.Client.Get(TestContext, key, latestNamespace); err != nil {
```

Comment:
As stated in the TODO, if `TestContext` is already closed this has no hope of succeeding. The cleanup seems like it isn't necessarily subject to the global test context depending on the... context

Reply:
Fixed